services/src/pro/datasets/postgres.rs

```diff
@@ -861,11 +912,631 @@ impl From<FileUpload> for crate::datasets::upload::FileUpload {
     }
 }
 
+#[async_trait]
+impl<Tls> UploadedUserDatasetStore for ProPostgresDb<Tls>
+where
+    Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static + std::fmt::Debug,
+    <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
+    <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
+    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
+{
+    async fn add_uploaded_dataset(
+        &self,
+        upload_id: UploadId,
+        dataset: AddDataset,
+        meta_data: MetaDataDefinition,
+    ) -> Result<DatasetIdAndName> {
+        let id = DatasetId::new();
+        let name = dataset.name.unwrap_or_else(|| DatasetName {
+            namespace: Some(self.session.user.id.to_string()),
+            name: id.to_string(),
+        });
+
+        log::info!(
+            "Adding dataset with name: {:?}, tags: {:?}",
+            name,
+            dataset.tags
+        );
+
+        if let Some(tags) = &dataset.tags {
+            if tags.contains(&"deleted".to_string()) {
```

Comment:
maybe we should have a reserved internal representation that users are not allowed to use themselves?
then again we don't do this for the special "upload" tag either...

Reply:
I added Reserved Tags enum and have log a warning at the handler level. Could throw an error there if desired. c287c3459b425802b5a54c2060145cbb3a955115